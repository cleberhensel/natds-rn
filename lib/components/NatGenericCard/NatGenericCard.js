import React from 'react';
import PropTypes from 'prop-types';
import { CustomView } from './NatGenericCard.styles';

export const NatGenericCard = ({
  children,
  height,
  width,
  bgColor,
}) => (
  <CustomView
    height={height}
    width={width}
    bgColor={bgColor}
    style={{
      shadowColor: '#000000', shadowOffset: { width: 0, height: 1 }, shadowOpacity: 0.14, shadowRadius: 1,
    }}
  >
    {children}
  </CustomView>
);


NatGenericCard.propTypes = {
  children: PropTypes.element.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
  bgColor: PropTypes.string,
};

NatGenericCard.defaultProps = {
  bgColor: '#FFFFFFFF',
  height: 216,
  width: 363,
};
