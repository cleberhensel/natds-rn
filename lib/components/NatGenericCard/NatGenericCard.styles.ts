import styled from 'styled-components/native';

export const CustomView = styled.View`
  elevation: 1;
  height: ${props => props.height};
  width: ${props => props.width}%;
  background-color: ${props => props.bgColor};
`;
