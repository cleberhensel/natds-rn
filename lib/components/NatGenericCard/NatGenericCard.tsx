import React from "react";
import { CustomView } from "./NatGenericCard.styles";
import { colors } from "../../assets/theme/colors/colors";
type NatGenericCardProps = {
  bgColor?: string,
  height?: number,
  width?: number
};
export const NatGenericCard: React.SFC<NatGenericCardProps> = ({
  children,
  height,
  width,
  bgColor
}) => (
  <CustomView height={height} width={width} bgColor={bgColor}>
    {children}
  </CustomView>
);
NatGenericCard.defaultProps = {
  bgColor: colors.primary.white,
  height: 216,
  width: 363
};
