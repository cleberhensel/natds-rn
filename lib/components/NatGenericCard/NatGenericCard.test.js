import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';
import { Text } from 'react-native';
import { NatGenericCard } from './NatGenericCard';

describe('NatGenericCard', () => {
  it('Should match snapshot', () => {
    const card = renderer.create(
      <NatGenericCard>
        <Text>Teste</Text>
      </NatGenericCard>,
    ).toJSON();
    expect(card).toMatchSnapshot();
  });

  describe('NatGenericCard with props - children', () => {
    it('Should render with correct children', () => {
      const component = renderer.create(
        <NatGenericCard>
          <Text>Teste</Text>
        </NatGenericCard>,
      ).toJSON();
      expect(component.children[0].type).toBe('Text');
    });
  });
});
