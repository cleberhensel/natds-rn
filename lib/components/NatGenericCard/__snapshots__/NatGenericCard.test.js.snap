// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatGenericCard Should match snapshot 1`] = `
<View
  bgColor="#FFFFFFFF"
  height={216}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFFFF",
        "elevation": 1,
        "height": 216,
        "width": 363,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.14,
        "shadowRadius": 1,
      },
    ]
  }
  width={363}
>
  <Text>
    Teste
  </Text>
</View>
`;
